pkg/apis/servicecatalog/types.go

```diff
@@ -127,10 +127,9 @@ type ServiceClass struct {
 	Plans         []ServicePlan
 	PlanUpdatable bool // Do we support this?
 
-	// OSB-specific
-	// OSBGUID is the identity of this object for use with the OSB API.
+	// ExternalID is the identity of this object for use with the OSB API.
 	// Immutable.
-	OSBGUID string
+	ExternalID string
```

Comment:
I just realized that my comment could be read more than one way. I didn't intend to mean that we would twiddle the name field, rather that we could use a different value than the ID once we could determine what other value we wanted the user to specify.

your idea is related to my question about virtual resources in some other comment. If kube has the concept of write-only properties, does it have the concept of read-only/generated properties?